packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -584,7 +597,29 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
             if (widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
             if (widget.header != null) widget.header,
             for (int i = 0; i < widget.children.length; i += 1) _wrap(widget.children[i], i, constraints),
-            if (!widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
+            if (widget.footer != null)
+              // We need to wrap the widget.footer so we can reorder an item between
+              // the last item of the list and the footer itself. If the footer
+              // is used instead of the finalDropArea it needs its own key,
+              // because _wrap() requires it.
+              _wrap(ConstrainedBox(
+                  key: endWidgetKey,
+                  constraints: BoxConstraints(
+                    minHeight: widget.scrollDirection == Axis.vertical ? _defaultDropAreaExtent : 0,
+                    minWidth: widget.scrollDirection == Axis.horizontal ? _defaultDropAreaExtent : 0,
+                  ),
+                  child: Align(
+                    alignment: footerAlignment,
+                    child: widget.footer,
+                  ),
+                ),
+                widget.children.length,
+                constraints,
+              ),
+            // We don't need a final drop area when a footer is present
```

Comment:
This isn't necessarily true and depends on the size of the footer widget (I think I mentioned this before). We should make sure that there is some heuristic or logic to figure out large the final drop area should be if there is a footer.

Otherwise, we have to at least be very clear to users that the footer has to be big enough to drag a list item, but I feel that this is could easily be a source of confusion